vpr/CMakeLists.txt

```diff
@@ -143,16 +145,19 @@ endif()
 #
 set(PROF_GEN_FLAGS_TO_CHECK
     #GCC-like
-    "-fprofile-generate" #Build will generate profiling information
+    "-fprofile-generate=${VPR_PGO_DATA_DIR}" #Build will generate profiling information
     )
 set(PROF_USE_FLAGS_TO_CHECK
     #GCC-like
-    "-fprofile-use"     #Build will use previously generated profiling information to guide code optimization
+    "-fprofile-use=${VPR_PGO_DATA_DIR}"     #Build will use previously generated profiling information to guide code optimization
     "-Wmissing-profile" #Warn if the used profiling information doesn't match the source code or is missing
     )
 
 if (VPR_PGO_CONFIG STREQUAL "prof_gen")
     message(STATUS "VPR: building to generate profiling data for PGO")
+    target_link_libraries(vpr gcov)
+    target_link_libraries(libvpr gcov)
```

Comment:
Shouldn't need to add gcov to both `libvpr` and `vpr`.  Just adding it to `libvpr` should be enough.

What is `CMAKE_REQUIRED_LIBRARIES` also required, versus just the `target_link_libraries` call?

Reply:
Without `CMAKE_REQUIRED_LIBRARIES`, the flag test fails and PGO is disabled. This is the only way I could find to add `-lgcov` to the test, which should be something like `$(CC) test.c -fprofile-generate -lgcov`.

Reply:
I've removed `target_link_libraries(vpr gcov)`.

Reply:
Is this a case where you want to use `find_library(GCOV_LIB gcov REQUIRED)`?  Is that what you are trying to accomplish with `CMAKE_REQUIRED_LIBRARIES`?

Reply:
No, it still fails with `find_library(GCOV_LIB gcov REQUIRED)`

```
-- VPR: building to generate profiling data for PGO
-- Performing Test CXX_COMPILER_SUPPORTS_-fprofile-generate=/tmp/pgo
-- Performing Test CXX_COMPILER_SUPPORTS_-fprofile-generate=/tmp/pgo - Failed
```
and in `CMakeError.log`:
```
g++   -DCXX_COMPILER_SUPPORTS_-fprofile-generate=/tmp/pgo    CMakeFiles/cmTC_996f9.dir/src.cxx.o  -o cmTC_996f9
ld: CMakeFiles/cmTC_996f9.dir/src.cxx.o: in function `main':
rect_call'
ld: src.cxx:(.text.startup+0x27): undefined reference to `__gcov_indirect_call_profiler_v3'
ld: src.cxx:(.text.startup+0x40): undefined reference to `__gcov_time_profiler_counter'
ld: CMakeFiles/cmTC_996f9.dir/src.cxx.o: in function `_GLOBAL__sub_I_00100_0_src.cxx':
t'
ld: CMakeFiles/cmTC_996f9.dir/src.cxx.o: in function `_GLOBAL__sub_D_00100_1_src.cxx':

ld: CMakeFiles/cmTC_996f9.dir/src.cxx.o:(.data.rel+0x20): undefined reference to `__gcov_merge_add'
ld: CMakeFiles/cmTC_996f9.dir/src.cxx.o:(.data.rel+0x58): undefined reference to `__gcov_merge_time_profile'
collect2: error: ld returned 1 exit status
make[1]: *** [CMakeFiles/cmTC_996f9.dir/build.make:87: cmTC_996f9] Error 1
```

Reply:
I think I understand what is happening here that I was missing before. Can you add a comment above this `set` that this is only affecting the `CHECK_CXX_COMPILER_FLAG` below.  I think with that comment future readers will get what is going on here.

Reply:
I've added a comment to clarify.